### PR TITLE
DS3231 support

### PR DESCRIPTION
**Pull request by bhuvanchandra**

### CODE REVIEW

File: extras/ds3231/ds3231.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#define DS3231_CTRL_SQWAVE_8192HZ 0x18
+#define DS3231_CTRL_SQWAVE_1HZ    0x00
+#define DS3231_CTRL_ALARM_INTS    0x04
+#define DS3231_CTRL_ALARM2_INT    0x02
+#define DS3231_CTRL_ALARM1_INT    0x01
+
+#define DS3231_ALARM_NONE 0
+#define DS3231_ALARM_1    1
+#define DS3231_ALARM_2    2
+#define DS3231_ALARM_BOTH 3
```
Comment (UncleRus) [inline extras/ds3231/ds3231.h]: It could be enums

File: extras/ds3231/ds3231.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#define DS3231_ALARM1_MATCH_SECMIN         2
+#define DS3231_ALARM1_MATCH_SECMINHOUR     3
+#define DS3231_ALARM1_MATCH_SECMINHOURDAY  4
+#define DS3231_ALARM1_MATCH_SECMINHOURDATE 5
+
+#define DS3231_ALARM2_EVERY_MIN         0
+#define DS3231_ALARM2_MATCH_MIN         1
+#define DS3231_ALARM2_MATCH_MINHOUR     2
+#define DS3231_ALARM2_MATCH_MINHOURDAY  3
+#define DS3231_ALARM2_MATCH_MINHOURDATE 4
```
Comment (UncleRus) [inline extras/ds3231/ds3231.h]: It could be enums

File: extras/ds3231/ds3231.c [hunk 3]
```diff
@@ -0,0 +1,346 @@
+#include "ds3231.h"
```
Comment (projectgus) [inline extras/ds3231/ds3231.c]: Could you add a copyright/license header here & on ds3231.h please? Doesn't have to be the full license, just a short one like for the example would be fine.

**Comment by projectgus** [comment]
Thanks for contributing this @bhuvanchandra . Looks good to me apart from the comment about copyright headers, although I agree with @UncleRus's comments about preferring enums over macros where possible.

File: extras/ds3231/ds3231.h (hunk 2 above)
Comment (bhuvanchandra) [inline extras/ds3231/ds3231.h]: Okay.

File: extras/ds3231/ds3231.c (hunk 3 above)
Comment (bhuvanchandra) [inline extras/ds3231/ds3231.c]: sure! will do it.

**Pull request closed without merging**